### ISSUE TITLE
[Train] Add support for Bagua

### ISSUE THREAD

**Issue by amogkam** [ray-project/ray#21934]
The Bagua library (https://github.com/BaguaSys/bagua) is optimized for high performant PyTorch training and is shown to have higher throughput than PyTorch DDP and Horovod. We can add support for it in Ray Train to make it very easy to run distributed training jobs with Bagua.

**Comment by jwyyy** [comment]
Hi @amogkam, I'd like to look into this issue 😄 if no one is working on this.

**Comment by amogkam** [comment]
Yes this would be a great project to work on! Though this is a little bit more complex, so maybe we put together an rfc first before starting with the implementation.

cc @matthewdeng

**Comment by cszhu** [comment]
Hello! This P2 issue has seen no activity in at least 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
